src/EditorFeatures/CSharpTest/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -5474,6 +5626,155 @@ void M()
     }
 
     int M2() => 0;
+}", options: PreferDiscard);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        [InlineData(nameof(PreferDiscard))]
+        [InlineData(nameof(PreferUnusedLocal))]
+        public async Task ValueOverwrittenByOutVar_ConditionalAndExpression(string optionName)
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        int {|FixAllInDocument:x1|} = -1, x2 = -1;
+        if (M2(x: out x1) &&
+            M2(x: out x2))
+        {
+            x1 = 0;
+            x2 = 0;
+        }
+        else
+        {
+            Console.WriteLine(x1);
+        }
+
+        Console.WriteLine(x1 + x2);
+    }
+
+    bool M2(out int x)
+    {
+        x = 0;
+        return true;
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        int x2 = -1;
+        int x1;
+        if (M2(x: out x1) &&
+            M2(x: out x2))
+        {
+            x1 = 0;
+            x2 = 0;
+        }
+        else
+        {
+            Console.WriteLine(x1);
+        }
+
+        Console.WriteLine(x1 + x2);
+    }
+
+    bool M2(out int x)
+    {
+        x = 0;
+        return true;
+    }
+}", optionName);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        [InlineData("var")]
+        [InlineData("int")]
+        public async Task UnusedOutVariableDeclaration_PreferDiscard(string typeName)
+        {
+            await TestInRegularAndScriptAsync(
+$@"class C
+{{
+    void M()
+    {{
+        if (M2(out {typeName} [|x|]))
+        {{
+        }}
+    }}
+
+    bool M2(out int x)
+    {{
+        x = 0;
+        return true;
+    }}
+}}",
+@"class C
+{
+    void M()
+    {
+        if (M2(out _))
+        {
+        }
+    }
+
+    bool M2(out int x)
+    {
+        x = 0;
+        return true;
+    }
+}", options: PreferDiscard);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task UnusedOutVariableDeclaration_MethodOverloads_PreferDiscard()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        if (M2(out int [|x|]))
+        {
+        }
+    }
+
+    bool M2(out int x)
+    {
+        x = 0;
+        return true;
+    }
+
+    bool M2(out char x)
+    {
+        x = 'c';
+        return true;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (M2(out int _))
+        {
+        }
+    }
+
+    bool M2(out int x)
+    {
+        x = 0;
+        return true;
+    }
+
+    bool M2(out char x)
+    {
+        x = 'c';
+        return true;
+    }
```

Comment:
wasn't there a case where we would end up with:

```c#
int x;
Foo(out _);
```

but we wanted to remove the 'x'?  Has that been done?  Was thre a test here that i missed?

Reply:
Yes, that was done in the previous PR before merge to master: https://github.com/dotnet/roslyn/pull/30777/commits/97424a343f354e6f3aed1f0c6000a9cda48c4953. See added test [here](https://github.com/dotnet/roslyn/pull/30777/commits/97424a343f354e6f3aed1f0c6000a9cda48c4953#diff-49e7f91d7855ff9b7f1e1d2f16439813R1516)

Reply:
Nice.  Thanks.  Sorry i missed that! :(